atc/engine/step_factory.go

```diff
@@ -77,7 +80,7 @@ func (factory *coreStepFactory) GetStep(
 		containerMetadata,
 		factory.lockFactory,
 		factory.resourceCacheFactory,
-		factory.strategy,
+		factory.noInputStrategy,
```

Comment:
Would we be able to make it so that if you don't pass in a `no-input-strategy` it will continue to use the `factory.strategy`? This way the addition of this new flag won't affect users that don't configure it.

Reply:
Changed noInputStrategy to default to strategy.